internal/project/root.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package project defines global project helpers.
+package project
+
+import (
+	"path/filepath"
+	"runtime"
+)
+
+var _, self, _, _ = runtime.Caller(0)
+
+// Root returns the filepath to the root of this project.
+func Root() string {
```

Comment:
This could be a global constant, like `self`, and the init generation code in the compiler will sort out the dependency chain.

Reply:
It'd be a global `var`, which would be mutable. AFAIK, you can't create a constant that uses `filepath.Join` since it wouldn't be constant across systems.

Reply:
fair point. `Root` could still be a LHS.

Reply:
What's an LHS?

Reply:
left hand side.